crates/biome_configuration/src/linter/rules.rs

```diff
@@ -67,24 +67,28 @@ impl DeserializableValidator for Rules {
 }
 impl Rules {
     #[doc = r" Checks if the code coming from [biome_diagnostics::Diagnostic] corresponds to a rule."]
-    #[doc = r" Usually the code is built like {category}/{rule_name}"]
-    pub fn matches_diagnostic_code<'a>(
-        &self,
-        category: Option<&'a str>,
-        rule_name: Option<&'a str>,
-    ) -> Option<(&'a str, &'a str)> {
-        match (category, rule_name) {
-            (Some(category), Some(rule_name)) => match category {
-                "a11y" => A11y::has_rule(rule_name).then_some((category, rule_name)),
-                "complexity" => Complexity::has_rule(rule_name).then_some((category, rule_name)),
-                "correctness" => Correctness::has_rule(rule_name).then_some((category, rule_name)),
-                "nursery" => Nursery::has_rule(rule_name).then_some((category, rule_name)),
-                "performance" => Performance::has_rule(rule_name).then_some((category, rule_name)),
-                "security" => Security::has_rule(rule_name).then_some((category, rule_name)),
-                "style" => Style::has_rule(rule_name).then_some((category, rule_name)),
-                "suspicious" => Suspicious::has_rule(rule_name).then_some((category, rule_name)),
-                _ => None,
-            },
+    #[doc = r" Usually the code is built like {group}/{rule_name}"]
+    pub fn matches_diagnostic_code(
+        group: &str,
+        rule_name: &str,
```

Comment:
Why was this changed?

Reply:
Because I need static string slices to avoid copying the rule code over and over. Previously the emitted string slices had a lifetime that depends on the function parameters.